src/PixiEditor/ViewModels/SubViewModels/Main/LayersViewModel.cs

```diff
@@ -48,10 +53,80 @@ public bool CanDeleteSelected()
     [Command.Basic("PixiEditor.Layer.DeleteSelected", "LAYER_DELETE_SELECTED", "LAYER_DELETE_SELECTED_DESCRIPTIVE", CanExecute = "PixiEditor.Layer.CanDeleteSelected", IconPath = "Trash.png")]
     public void DeleteSelected()
     {
-        var member = Owner.DocumentManagerSubViewModel.ActiveDocument?.SelectedStructureMember;
+        var doc = Owner.DocumentManagerSubViewModel?.ActiveDocument;
+
+        var member = doc.SelectedStructureMember;
         if (member is null)
             return;
+
         member.Document.Operations.DeleteStructureMember(member.GuidValue);
+        var allLayers = doc.StructureHelper.GetAllMembers().ToArray();
```

Comment:
you can iterate over `doc.StructureRoot.Children` directly, no need to copy the structure into a separate list. You can get rid of both `GetAllMembers` overloads (one of them seems to be unused already?)

Edit: although, you likely won't need `doc.StructureRoot.Children` at all, see further comments